File: Python/ProgramStudi.py
from Course import Course

class ProgramStudi:
    __namaProdi = ""
    __kodeProdi = ""
    __listCourse = []

    def __init__(self):
        self.namaProdi =""
        self.kodeProdi = ""
        self.listCourse = []
    
    def setNamaProdi(self, a):
        self.namaProdi = a
    def setKodeProdi(self, a):
        self.kodeProdi = a
    def addCourse(self, nm, sks):
        temp = Course()
        temp.setNamaMatkul(nm)
        temp.setJumlahSKS(sks)
        self.listCourse.append(temp)

    def getNamaProdi(self):
        return self.namaProdi
    def getKodeProdi(self):
        return self.kodeProdi
    def getCourse(self):
        return self.listCourse

    

File: CPP/ProgramStudi.cpp
#ifndef PRODI
#define PRODI
#include <iostream>
#include <string>
#include <list>
#include "Course.cpp"

using namespace std;

class ProgramStudi
{
private:
    string kodeProdi;
    string namaProdi;
    list<Course> listCourse;

public:
    ProgramStudi()
    {
        kodeProdi = "";
        namaProdi = "";
    };
    ProgramStudi(string a, string b)
    {
        kodeProdi = a;
        namaProdi = b;
    }

    void setkodeProdi(string a)
    {
        kodeProdi = a;
    }
    void setnamaProdi(string a)
    {
        namaProdi = a;
    }
    void addCourse(string a, int b)
    {
        Course temp;
        temp.setNamaMatkul(a);
        temp.setJumlahSKS(b);
        listCourse.push_back(temp);
    }

    string getkodeProdi()
    {
        return kodeProdi;
    }
    string getnamaProdi()
    {
        return namaProdi;
    }
    list<Course> getCourse()
    {
        return listCourse;
    }
    ~ProgramStudi()
    {
    }
};
#endif